android/src/main/kotlin/com/apptentive/apptentive_flutter/ApptentiveFlutterPlugin.kt

```diff
@@ -286,74 +302,108 @@ class ApptentiveFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware
+                "Person Name: ${notification?.personName}. " +
+                "Person Email: ${notification?.personEmail}"
+
+      Log.d(LogTags.MESSAGE_CENTER_NOTIFICATION, notificationText)
+      if (notification?.unreadMessageCount != 0) {
+        android.util.Log.d("TESTING...........", "calling onUnreadMessageCountChanged")
+        activity?.runOnUiThread {
+          channel.invokeMethod("onUnreadMessageCountChanged", mapOf("count" to notification?.unreadMessageCount))
+        }
+      }
+    }
+  }
 
-  ////////// UTIL METHODS
-
-
-
-  // Parse log level string into Apptentive LogLevel
-  // Return ApptentiveLog.Level.INFO by default
-  private fun parseLogLevel(logLevelStr: String): ApptentiveLog.Level {
-    if (logLevelStr.contains("verbose")) return ApptentiveLog.Level.VERBOSE
-    if (logLevelStr.contains("debug")) return ApptentiveLog.Level.DEBUG
-    if (logLevelStr.contains("info")) return ApptentiveLog.Level.INFO
-    if (logLevelStr.contains("warn")) return ApptentiveLog.Level.WARN
-    if (logLevelStr.contains("error")) return ApptentiveLog.Level.ERROR
-    Log.e("Apptentive Flutter", "Unknown log level: $logLevelStr. Log level is set to .INFO by default")
-    return ApptentiveLog.Level.INFO
+  // endregion
+
+  // region Utils
+
+  private fun parseLogLevel(logLevelStr: String): LogLevel {
+    return when {
+        logLevelStr.contains("verbose") -> LogLevel.Verbose
+        logLevelStr.contains("debug") -> LogLevel.Debug
+        logLevelStr.contains("info") -> LogLevel.Info
+        logLevelStr.contains("warn") -> LogLevel.Warning
+        logLevelStr.contains("error") -> LogLevel.Error
+        else -> {
+          Log.w(LogTag("Flutter LogLevel"),
+            "Unknown log level: $logLevelStr. Log level is set to .INFO by default")
+          LogLevel.Info
+        }
+    }
   }
 
   // Turn a Map into an Apptentive Configuration
-  // Suppress unchecked cast from Any? to Map<String,String>?, since we know it will be that type
   @Suppress("UNCHECKED_CAST")
   private fun unpackConfiguration(configurationMap: Map<String, Any>): ApptentiveConfiguration {
-    // Key/Sig
     val key = configurationMap["key"] as String
     val signature = configurationMap["signature"] as String
-    // Optional parameters
     val logLevel = parseLogLevel(configurationMap["log_level"] as String)
+    val shouldInheritAppTheme = configurationMap["should_inherit_theme"] as Boolean
     val shouldEncryptStorage = configurationMap["should_encrypt_storage"] as Boolean
     val shouldSanitizeLogMessages = configurationMap["should_sanitize_log_messages"] as Boolean
-    val troubleshootingModeEnabled = configurationMap["troubleshooting_mode_enabled"] as Boolean
-    // val shouldCollectAndroidIdOnPreOreoTargets = configurationMap["should_collect_android_id_on_pre_oreo_targets"] as Boolean
-    val surveyTermsAndConditions: TermsAndConditions? = unpackTermsAndConditions(configurationMap["terms_and_conditions"] as Map<String, String>?)
```

Comment:
@frankus I want to confirm iOS doesn't need surveyTermsAndConditions. I thought you all were using it.